src/managers/UpdateManager.ts

```diff
@@ -139,6 +141,24 @@ export class UpdateManager {
     /* tslint:enable:no-floating-promises */
 
     // 2. Update push player with external_user_id
+    const pushUpdatePromise = this.sendPushPlayerUpdate(payload);
+    if (await this.isDeviceIdAvailable()) {
+      await pushUpdatePromise;
+    }
+  }
+
+  // Send REST API payload to update the push player record.
+  // Await until we have a push playerId, which is require to make a PUT call.
+  private async sendPushPlayerUpdate(payload: UpdatePlayerOptions): Promise<void> {
+    let deviceId =  (await Database.getSubscription()).deviceId;
```

Comment:
I believe you can destructure here

Reply:
We can, but was trying to keep it the same as line 156 below so it is easier to spot we are doing the same thing. Thoughts?

Reply:
It seems the deconstruct syntax does work with reassignment, but requires an extra set of parentheses.
```typescript
    let { deviceId } =  await Database.getSubscription();
    if (!deviceId) {
      await awaitSdkEvent(OneSignal.EVENTS.REGISTERED);
      ({ deviceId } = await Database.getSubscription());
    }
```

This looks a bit better so I'll update it to this.

Reply:
Done

Reply:
Ah interesting. I didn't know that one.